pkg/debuginfo/client.go

```diff
@@ -21,17 +21,26 @@ import (
 	"google.golang.org/grpc"
 )
 
-type NoopClient struct{}
+type NoopClient struct {
+	ShouldInitiateUploadF func(in *debuginfopb.ShouldInitiateUploadRequest, opts ...grpc.CallOption) (*debuginfopb.ShouldInitiateUploadResponse, error)
```

Comment:
Rather than having these functions as part of `NoopClient,` we could implement a client for testing in the test file. That way, it would be easier to follow. What do you think?

Reply:
That's a good idea 👍

Reply:
I've decoupled a test client from `NoopClient`. It can be made exportable later on if some package needs it.